zio-http/src/test/scala/zio/http/endpoint/EndpointSpec.scala

```diff
@@ -302,6 +302,59 @@ object EndpointSpec extends ZIOSpecDefault {
           testRoutes("/users/123/posts/555?name=adam", Method.PUT)
         },
       ),
+      suite("custom error")(
+        test("simple custom error response") {
+          val routes =
+            Endpoint
+              .get(literal("users") / int("userId"))
+              .out[String]
+              .outError[String](Status.Custom(999))
+              .implement { userId =>
+                ZIO.fail(s"path(users, $userId)")
+              }
+
+          val request =
+            Request
+              .get(
+                URL.decode("/users/123").toOption.get,
+              )
+
+          for {
+            response <- routes.toApp.runZIO(request).mapError(_.get)
+            body     <- response.body.asString.orDie
+          } yield assertTrue(response.status.code == 999, body == "\"path(users, 123)\"")
+        },
+        test("status depending on the error subtype") {
+          val routes =
+            Endpoint
+              .get(literal("users") / int("userId"))
+              .out[String]
+              .outErrors[TestError, TestError.UnexpectedError, TestError.InvalidUser](
```

Comment:
I wonder if we can improve this to:

```suggestion
              .outErrors[TestError](
```

by using partial application trick (intermediate class with an `apply`)?

Reply:
Done (both here and in the `HttpCodec.enumeration`)